microxrce_agent.cpp

```diff
@@ -169,23 +240,51 @@ int main(int argc, char** argv)
                 tty_config.c_cc[VMIN] = 10;
                 tty_config.c_cc[VTIME] = 1;
 
+                /*
+                 * Parse Baudrate.
+                 */
+                speed_t baudrate = B115200;
+                if ((cl_counter <= cl.size()) && ("--baudrate" == cl[cl_counter -1]))
+                {
+                    cl_counter++;
+                    if (cl_counter <= cl.size())
+                    {
+                        baudrate = getBaudRate(cl[cl_counter - 1].c_str());
+                        cl_counter++;
+                    }
+                }
+
                 /* Setting BAUD RATE. */
-                speed_t baudrate = (3 == cl.size()) ? getBaudRate(cl[2].c_str()) : B115200;
-                if (0 != baudrate)
+                cfsetispeed(&tty_config, baudrate);
+                cfsetospeed(&tty_config, baudrate);
+
+                if (0 == tcsetattr(fd, TCSANOW, &tty_config))
                 {
-                    cfsetispeed(&tty_config, baudrate);
-                    cfsetospeed(&tty_config, baudrate);
+                    server.reset(new eprosima::uxr::SerialServer(fd, 0));
+                }
 
-                    if (0 == tcsetattr(fd, TCSANOW, &tty_config))
+                /*
+                 * Parse Refs File.
+                 */
+                if ((cl_counter <= cl.size()) && ("--refs" == cl[cl_counter - 1]))
+                {
+                    cl_counter++;
+                    if (cl_counter <= cl.size())
+                    {
+                        server->load_config_file(cl[cl_counter -1]);
```

Comment:
Shall we do something with the returned bool?

Reply:
Fixed!